src/components/TableDialog.vue

```diff
@@ -111,130 +111,125 @@
 </template>
 
 <script lang="ts">
-import Vue, { PropType } from 'vue';
+import {
+  defineComponent, ref, Ref, computed,
+} from '@vue/composition-api';
 
 import api from '@/api';
 import { FileType, CSVColumnType } from '@/types';
 import { validFileType, fileName as getFileName, csvFileTypeRecommendations } from '@/utils/files';
 
 const defaultKeyField = '_key';
-export default Vue.extend({
-  name: 'TableDialog',
+const fileTypes: readonly FileType[] = [
+  {
+    extension: ['csv'],
+    queryCall: 'csv',
+    hint: 'Comma Separated Value file',
+    displayName: 'CSV',
+  },
+];
 
+export default defineComponent({
+  name: 'TableDialog',
   props: {
     workspace: {
-      type: String as PropType<string>,
+      type: String,
       required: true,
     },
   },
-  data() {
-    return {
-      tableDialog: false,
-      types: [
-        {
-          extension: ['csv'],
-          queryCall: 'csv',
-          hint: 'Comma Separated Value file',
-          displayName: 'CSV',
-        },
-      ] as FileType[],
-      file: null as File | null,
-      fileName: null as string | null,
-      fileUploadError: null as string | null,
-      tableCreationError: null as string | null,
-      key: defaultKeyField,
-      overwrite: false,
-      columnType: {} as { [key: string]: CSVColumnType },
-      uploading: false,
-      uploadProgress: null as number | null,
-    };
-  },
-  computed: {
-    createDisabled(): boolean {
-      return (
-        !this.file
-        || !this.fileName
-        || !!this.fileUploadError
-      );
-    },
-
-    multinetTypes(): CSVColumnType[] {
-      return [
-        'label',
-        'category',
-        'number',
-        'date',
-      ];
-    },
-  },
-
-  methods: {
-    restoreKeyField() {
-      this.key = defaultKeyField;
-    },
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async handleFileInput(this: any, file: File | undefined) {
-      this.file = file || null;
+  setup(props, { emit }) {
+    // Type recommendation
+    const columnType: Ref<{[key: string]: CSVColumnType}> = ref({});
+    const multinetTypes: readonly CSVColumnType[] = ['label', 'category', 'number', 'date'];
+
+    // File selection
+    const selectedFile = ref<File | null>(null);
+    const fileName = ref<string | null>(null);
+    const fileUploadError = ref<string | null>(null);
+    async function handleFileInput(file: File) {
```

Comment:
Apparently this is allowed to be `undefined` in line 152 (going by the code this line replaced). I think the type of `file` should be `file | undefined`, and the test in 152 should probably be explicit for `undefined`.

(Feel free to leave the test in 152 as it is, since it was like that before. But the type should definitely be changed, and if you changed the test to be more explicit, I think that would actually be an improvement.)

Reply:
You're right, good catch. I'll fix it.